### PR TITLE
feat(Reordering): Add nim reordering methods

### PR DESCRIPTION
**Pull request by borismelnik**
https://github.com/status-im/StatusQ/pull/300 and https://github.com/status-im/status-go/pull/2301 are required

### CODE REVIEW

**Review by jrainville — APPROVED** [review]

**Review by pinebit — APPROVED** [review]

**Review by iurimatias — CHANGES_REQUESTED** [review]
doesn't work for me, see video


https://user-images.githubusercontent.com/176720/128252557-b7a38e29-fc01-4007-bd78-25b39be045a5.mov

File: src/status/signals/messages.nim [hunk 1]
```diff
@@ -157,6 +157,8 @@ proc toChat*(jsonChat: JsonNode): Chat =
   let chatTypeInt = jsonChat{"chatType"}.getInt
   let chatType: ChatType = if chatTypeInt >= ord(low(ChatType)) or chatTypeInt <= ord(high(ChatType)): ChatType(chatTypeInt) else: ChatType.Unknown
 
+  echo $jsonChat
```
Comment (richard-ramos) [inline src/status/signals/messages.nim:160]:
```suggestion
```

File: src/status/signals/messages.nim [hunk 2]
```diff
@@ -174,7 +176,8 @@ proc toChat*(jsonChat: JsonNode): Chat =
     muted: false,
     ensName: "",
     joined: 0,
-    private: jsonChat{"private"}.getBool
+    private: jsonChat{"private"}.getBool,
+    position: jsonChat{"position"}.getInt
```
Comment (richard-ramos) [inline src/status/signals/messages.nim:180]:
```suggestion
```
`jsonChat` does not have a `position`. It's part of the chat information of the communities. Should be added here:

https://github.com/status-im/status-desktop/blob/master/src/status/signals/messages.nim#L249-L260

**Review by richard-ramos — CHANGES_REQUESTED** [review]
Added some comments, also please change the commit used in vendor/status-go to point to https://github.com/status-im/status-go/pull/2301

File: src/status/signals/messages.nim (hunk 2 above)
Comment (borismelnik) [reply src/status/signals/messages.nim:179]: already remove it just no pushed

**Comment by PascalPrecht** [comment]
Closing this in favour of https://github.com/status-im/status-desktop/pull/3257

**Pull request closed without merging**